oauthproxy.go

```diff
@@ -980,6 +990,9 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 
 // addHeadersForProxying adds the appropriate headers the request / response for proxying
 func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) {
+	for _, extraHeader := range p.extraHeaders {
+		req.Header[extraHeader[0]] = []string{extraHeader[1]}
```

Comment:
Use `req.Header.Add` otherwise you will overwrite any existing values, unless that is intended? In which case, what happens if I pass `foo: bar`, `foo: baz` to the extra-headers option? That should come out as `foo: bar,baz` according to the HTTP spec

Reply:
Ok. I had just copied the surrounding code. Will switch to add.